examples/Azure_IoT_Hub_ESP32/Azure_IoT_Hub_ESP32.ino

```diff
@@ -296,15 +297,10 @@ static void establishConnection()
 
 static void getTelemetryPayload(az_span payload, az_span* out_payload)
 {
-  az_span original_payload = payload;
+  // You can generate the json using any lib you want. Here it's hardcoded for simplicity.
+  myTelemetry = "{ \"msgCount\": "+ String(telemetry_send_count++) + " }";
 
-  payload = az_span_copy(
-      payload, AZ_SPAN_FROM_STR("{ \"msgCount\": "));
-  (void)az_span_u32toa(payload, telemetry_send_count++, &payload);
-  payload = az_span_copy(payload, AZ_SPAN_FROM_STR(" }"));
-  payload = az_span_copy_u8(payload, '\0');
-
-  *out_payload = az_span_slice(original_payload, 0, az_span_size(original_payload) - az_span_size(payload));
+  *out_payload = az_span_create((uint8_t*)myTelemetry.c_str(), myTelemetry.length());
 }
```

Comment:
What library are you using String from?  That is a C++ type.  C uses c-strings (char arrays).

Reply:
So, Arduino use C++ and it's fine to use string